logger.go

```diff
@@ -0,0 +1,50 @@
+package llog
+
+import (
+	"log"
+	"strings"
+)
+
+type llogWriter struct {
+	fn      LogFunc
+	kv      KV
+	filters []func(string) (string, error)
+}
+
+// Write implements the io.Writer interface
+func (lw *llogWriter) Write(b []byte) (int, error) {
+	msg := strings.TrimSpace(string(b))
+	for _, fn := range lw.filters {
+		m, err := fn(msg)
+		if err != nil {
+			return 0, err
+		}
+		// ignore if there's no message left
+		if m == "" {
+			// pretend we still wrote everything
+			return len(b), nil
+		}
+		msg = m
+	}
+	// TODO: what should we do with multi-line messages
+	lw.fn(string(msg), lw.kv)
+	return len(b), nil
+}
+
+// NewLogger returns an instance of log.Logger that uses llog to log the
+// messages under the sent level with the passed KV. Multiple filter functions
+// can be passed. If an error is returned from the filter function, it's sent
+// to the caller of the Write method. If an empty string is returned then the
+// message is ignored.
+func NewLogger(lvl LogFunc, kv KV, filters ...func(string) (string, error)) *log.Logger {
```

Comment:
This is confusing because the var is lvl but it doesn't take one of the Level constants and it seems like this argument is actually ignored?